master/internal/experiment.go

```diff
@@ -284,6 +284,8 @@ func (e *experiment) Receive(ctx *actor.Context) error {
 		}
 		telemetry.ReportExperimentStateChanged(ctx.Self().System(), e.db, *e.Experiment)
 
+		ctx.Tell(e.hpImportance, hpimportance.ExperimentCompleted{ID: e.ID})
+
```

Comment:
Didn't you hit a panic before, that wouldn't be prevent by this re-ordering?

Reply:
I just tested changing the e.Searcher.shutdown conditional to `if e.State == model.CompletedState {` and it seems to work well, but I don't know exactly what you do to hit all the panics you did.

Reply:
I think it should be handled by https://github.com/determined-ai/determined/pull/2230/files#diff-f8f79514e8276c806fcfa81745d24ada457e637648894ebf1d006ac5d4a29e46R188 now.

>  just tested changing the e.Searcher.shutdown conditional to if e.State == model.CompletedState { and it seems to work well, but I don't know exactly what you do to hit all the panics you did.

That is super strange..